File: HatForFilms/ClientApp/src/Repository/FilmRepository.js
import { mainRepository } from './../Repository/mainRepository'

export class FilmRepository extends mainRepository {
    async getAll() {
        return await this.request('api/Films/GetAll', true, null, false, null);
    }

    async createNew(name) {
        let params = [];
        params.push({ name: 'filmName', val: name });
        return await this.request('api/Films/AddNew', false, params, true, null);
    }

    async searchByName(name) {
        let params = [];
        params.push({ name: 'filmName', val: name });
        return await this.request('api/Films/Find', true, params, false, null);
    }

    async updateFilm(film) {
        return await this.request('api/Films/Update', false, null, true, film);
    }
}

File: Models/Repositories/FilmRepository.cs
using Models.Classes;
using Models.Classes.Request;
using Models.Data;
using System.Collections.Generic;
using System.Linq;

namespace Models.Repositories
{
    public class FilmRepository
    {
        public List<Film> GetAll()
        {
            var fms = new List<Film>();

            using (var db = new ApplicationContext())
            {
                fms = db.Films.ToList();
            }

            return fms;
        }

        public bool CreateNewFilm(Film film)
        {
            try
            {
                using (var db = new ApplicationContext())
                {
                    db.Films.Add(film);
                    db.SaveChanges();
                }
                return true;
            }
            catch
            {
                return false;
            }
        }
        public bool Update(filmUpdateRequest filmUpdate)
        {
            try
            {
                using (var db = new ApplicationContext())
                {
                    var film = db.Films.FirstOrDefault(x=>x.Id == filmUpdate.Id);
                    film.Name = filmUpdate.Name;
                    film.Description = filmUpdate.Description;
                    db.Films.Update(film);
                    db.SaveChanges();
                }
                return true;
            }
            catch
            {
                return false;
            }
        }
    }
}
